pkg/network/usm/monitor.go

```diff
@@ -225,20 +241,50 @@ func (m *Monitor) Start() error {
 			m.kafkaProcess,
 		)
 		if err != nil {
-			return err
+			log.Errorf("could not enable kafka monitoring: %s", err)
+		} else {
+			m.kafkaConsumer.Start()
+			enabledCount++
 		}
-		m.kafkaConsumer.Start()
+	}
+
+	// No protocols could be enabled, abort.
+	if enabledCount == 0 {
+		return errNoProtocols
 	}
 
 	err = m.ebpfProgram.Start()
 	if err != nil {
 		return err
 	}
 
+	for protocolType, protocol := range m.protocols {
+		err = protocol.PostStart(m.ebpfProgram.Manager.Manager)
+		if err != nil {
+			// Cleanup the protocol
+			enabledCount--
+			protocol.PreStop(m.ebpfProgram.Manager.Manager)
+			delete(m.protocols, protocolType)
+
+			// Log and reset the error value
+			log.Errorf("could not complete post-start phase of %s monitoring: %s", protocolType, err)
+			err = nil
```

Comment:
why?

Reply:
Actually, I should have used another error variable for those startup errors than `err` here.

As to why it was needed if reusing `err`, it is because it is possible that `err` is never set again after that last loop of the `Start` method, and it is the return value of the method. If we look at the call of the `Start` method, we can see that if `Start` returns an error, then the monitor is dropped and the whole of USM is considered as having failed to start completely. Since we are trying to do best effort when initialising USM, one protocol failing does not necessarily mean we completely failed to initialise if there is at least one other protocol that succeeded its initialisation, and in that case we want to return a nil value for the error.